pkg/kubectl/cmd/apply_view.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/kubernetes/pkg/api/annotations"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	applyViewLong = templates.LongDesc(`
+		Get the last-applied-configuration annotations by type/name or file.
+
+		You can use --output jsonpath={...} to extract specific values using a jsonpath expression.`)
+
+	applyViewExample = templates.Examples(`
+		# Get the last-applied-configuration annotations by type/name
+		kubectl apply view last-applied deployment/nginx
+
+		# Get the last-applied-configuration annotations by file
+		kubectl apply view last-applied -f deployment_nginx.yaml -o json`)
+
+	lastApplied = "last-applied"
+)
+
+func NewCmdApplyView(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	options := &ApplyOptions {}
+	cmd := &cobra.Command{
+		Use:     "view (TYPE [NAME | -l label] | TYPE/NAME | -f FILENAME)",
+		Short:   "Get the last-applied-configuration annotations",
+		Long:    applyViewLong,
+		Example: applyViewExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := ApplyViewLastApplied(f, cmdOut, cmd, args, options)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "that contains the last-applied-configuration annotations"
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+	return cmd
+}
+
+func ApplyViewLastApplied(f cmdutil.Factory, errOut io.Writer, cmd *cobra.Command, args []string, options *ApplyOptions) error {
+	if args[0] != lastApplied {
+		return cmdutil.UsageError(cmd, "for now apply %s only support last-applied")
+	}
+
+	mapper, typer, err := f.UnstructuredObject()
+	if err != nil {
+		return err
+	}
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
```

Comment:
if we explicit flag `-n` or `--namespace`. then the `DefaultNamspace()` will return the value we set to `cmdNamespace`, sorry, I don't quit understand what do you mean by respect the namespace flag if set ?
btw, do we need to also support `all-namespace` and `selector`?